### PR TITLE
Verify --version output matches CHANGELOG

### PR DESCRIPTION
**Pull request by sbc100**
The new version string looks like this:
    
  wasm-opt version 90 (version_90-18-g77329439d)
    
The version reported here is the version from the CMakeLists.txt file
followed by the git version in brackets.  We verify that the main
version here matches the CHANGELOG to prevent people from changing
 one without changeing the other.

This will help with emscripten that wants to be able to programaticaly
check the --version of binaryen tools.

See https://github.com/emscripten-core/emscripten/issues/10175

### CODE REVIEW

**Comment by sbc100** [comment]
No sorry, I should clarify.   The new version string looks like this:

```
$ wasm-opt --version
wasm-opt version 90 (version_90-18-g77329439d)
```

So the git tag stuff all lives inside the brackets, and we don't check that.  We only check the cmake version which is part of the cmake file.

So this test will prevent anyone from updateing the changelog without remembering to update the CMakeLists.txt version too.  Which IMHO is exactly what we want. 

This format also match what llvm does:

```
$ ../llvm-build/bin/clang -v
clang version 10.0.0 (https://github.com/llvm/llvm-project.git 26ba160d47220a0bce75b1f491bf6e262edf69fa)
```

**Comment by sbc100** [comment]
So the git tags are basically irrelevant to the test, and to what emscripten will check.    Which is good because some people don't have a git checkout at all (they build from a source archive).

**Comment by sbc100** [comment]
I'm actually really happy with this solution.  It seems pretty bullet proof to me.

We could just drop the git version string completely if you prefer?

File: CMakeLists.txt [hunk 1]
```diff
@@ -15,17 +15,17 @@ option(BYN_ENABLE_ASSERTIONS "Enable assertions" ON)
...
+               GIT_VERSION_RESULT
            OUTPUT_VARIABLE
-               GIT_HASH
+               GIT_VERSION
            OUTPUT_STRIP_TRAILING_WHITESPACE)
-  if(${GIT_HASH_RESULT})
+  if(${GIT_VERSION_RESULT})
     message(WARNING "Error running git describe to determine version")
-    set(CMAKE_PROJECT_VERSION "(unable to determine version)")
+    set(CMAKE_PROJECT_VERSION "${CMAKE_PROJECT_VERSION} (unable to determine version)")
```
Comment (kripken) [inline CMakeLists.txt]: will this print `version 90 (unable to determine version)`? that seems confusing if so.

Comment (sbc100) [reply CMakeLists.txt]: Ah yes.  I think we can just hard error here since we are already wrapping this block in `if exists(.git)`

**Pull request merged**